src/OpenTelemetry/Metrics/Metric.cs

```diff
@@ -110,6 +110,13 @@ public sealed class Metric
             this.InstrumentDisposed = false;
         }
 
+        private Metric(Metric other)
+        {
+            this.MetricType = other.MetricType;
+            this.InstrumentIdentity = other.InstrumentIdentity;
+            this.aggStore = other.aggStore.Copy();
```

Comment:
One concern with this approach. AggregatorStore holds onto a pretty sizeable buffer (sizeof(MetricPoint) * 2000 essentially) so user might OOM if their inmemoryexporter is running long enough?

Reply:
good point!
It also looks like there's a circular reference with `AggregatorStore` appearing on both `Metric` and `MetricPoint`.